### PR TITLE
Is1793/serving multiple products

### PR DESCRIPTION
**Pull request by pcrespov**
## What do these changes do?

- serves front-end applications from PR #1788 based on host
    - temporary hard-coded all front-end outputs as products
    - defaults to osparc
- middleware that discovers and stores in request product key
- requests from webserver->catalog includes product in the headers

## Related issue number

#1793 

## How to test

In development mode, use the following links
- http://127.0.0.1:9081 -> osparc
- http://127.0.0.1:9081/osparc/index.html  -> osparc
- http://127.0.0.1:9081/s4l/index.html    -> s4l
- http://127.0.0.1:9081/tis/index.html  -> tis

## Checklist

- [x] Did you change any service's API? Then make sure to bundle document and upgrade version (``make openapi-specs``, ``git commit ...`` and then ``make version-*``)
- [x] Unit tests for the changes exist
- [x] Runs in the swarm
- [x] Documentation reflects the changes
- [x] New module? Add your github username to [.github/CODEOWNERS](.github/CODEOWNERS)

### CODE REVIEW

File: services/web/server/src/simcore_service_webserver/products.py [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+
+from servicelib.application_setup import ModuleCategory, app_module_setup
+
+from .__version__ import api_vtag
+from .constants import RQ_PRODUCT_KEY
+
+log = logging.getLogger(__name__)
+
+# TODO: <--- This is defined by services/web/client/compile.json
+FE_APPS = ["osparc", "s4l", "tis", "explorer", "apiviewer", "testtapper"]
```
Comment (odeimaiz) [inline services/web/server/src/simcore_service_webserver/products.py:15]: "explorer", "apiviewer", "testtapper" were removed

**Review by odeimaiz — APPROVED** [review]

File: services/web/server/src/simcore_service_webserver/products.py (hunk 1 above)
Comment (sanderegg) [reply services/web/server/src/simcore_service_webserver/products.py:15]: so I guess these will go in the database instead of hard-coded..

**Review by sanderegg — APPROVED** [review]
still needs injection of products in the database. or the catalog won't be happy.

File: services/web/server/src/simcore_service_webserver/products.py (hunk 1 above)
Comment (odeimaiz) [inline services/web/server/src/simcore_service_webserver/products.py:15]:
```suggestion
FE_APPS = ["osparc", "s4l", "tis"]
```

Comment (pcrespov) [reply services/web/server/src/simcore_service_webserver/products.py]: i know, it is updatedn and there is also a test that makes sure that the correspnoding front-end apps exist

**Pull request merged**